### DISCUSSION TITLE
Making webpack 5 name chunks as per webpack 4

### DISCUSSION THREAD

**Discussion by davea38** [webpack/webpack#15138] [Q&A] [answered]
Hi!

Webpack 4 used to name chunks x.chunk.js / x.chunk.css (starting at 0 and continuing to the highest chunk).

How can we re-create the same behaviour in Webpack 5? 
I've tried overriding the config.output.chunkFilename and the corresponding MiniCssExtractPlugin plugin options.chunkFilename, but can't seem to work it out...

I want to have two chunks, I would like them to be explicitly named:

0.chunk.js / 0.chunk.css
1.chunk.js / 1.chunk.css

This is for an integration with some powershell scripts that provision a Sharepoint 2013 site and expect the chunks to be statically named.

**Comment by alexander-akait** [comment]
You can use function for the `chunkFilename` option and create any logic, yes, we write it for better deterministic logic

**Reply by davea38** [reply]
I'm not sure how I would do this - would my function have to implement a static counter per chunk file? I also need to account for css chunking via MiniCssExtractPlugin and I'm not sure how to do this

**Comment by davea38** [comment]
I sorted it - for anyone who comes across this this was my solution

```javascript
const chunkNameManager = () => {
  const chunks = [];

  const calcChunkName = (chunk, ext = 'js') => {
    const chunkIndex = chunks.findIndex((c) => c.id === chunk.id && c.hash === chunk.hash);
    const chunkExists = chunkIndex > -1;

    let name = getChunkName(chunk.name, ext);

    // If there is no chunk name
    if (!chunk.name) {
      if (chunkExists) {
        name = getChunkName(chunkIndex, ext);
      } else {
        name = getChunkName(chunks.length, ext);
        // Add to the array
        chunks.push(chunk);
      }
    }

    return name;
  };

  return {
    calcChunkName,
  };
};
```

```javascript
const cssChunkManager = utils.chunkNameManager();

newConfig.plugins.map((plugin) => {
        // Configure MiniCssExtractPlugin to correctly name chunk file css
        if (plugin instanceof MiniCssExtractPlugin) {
          Object.assign(plugin.options, {
            filename: (pathData) => {
              const { chunk } = pathData;

              // [name].chunk.css | [x].chunk.css
              return cssChunkManager.calcChunkName(chunk, 'css');
            },
          });
        }

const jsChunkManager = utils.chunkNameManager();

newConfig.output = {
        ...newConfig.output,
        filename: (pathData) => {
          const { chunk } = pathData;

          // [name].chunk.js | [x].chunk.js
          return jsChunkManager.calcChunkName(chunk, 'js');
        },
      };
```

This was done in craco.config.js webpack.configure

**Comment by sokra** [comment, marked as answer]
`optimization.chunkIds: "size"` is the old behavior

**Reply by davea38** [reply]
Thank you :)
